src/fastoad/notebooks/03_custom_modules/beam_problem.ipynb

```diff
@@ -0,0 +1,271 @@
+    "* Then, your problem is defined within the `model` section. \n",
+    "* If variables are shared between different modules (models) then loops occurs and `nonlinear_solver` and `linear_solver` must be defined, here non linear Block Gauss-Seidel and linear Direct solvers are used. \n",
+    "The configuration file for the first problem is shown below. As you can notice, three different modules are called sequentially: the geometry, the weight and the displacements. As Displacements and geometry shares inputs/outputs we need to introduce solvers. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"row\">\n",
+    "  <div class=\"column\">\n",
+    "    <img src=\"./img/disp_config.gif\" width=\"800\">\n",
+    "  </div>\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once your configuration file is generated, you can check your models connections using the n2 diagram: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "oad.write_n2(CONFIGURATION_FILE_NAME, overwrite=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"row\">\n",
+    "  <div class=\"column\">\n",
+    "    <img src=\"./img/partition_tree_n2.svg\" width=\"800\">\n",
+    "  </div>\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Input file \n",
+    "\n",
+    "One the configuration is generated you can generate or use an already generated input file: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# UNCOMMENT THE FOLLOWING LINE TO GENERATE A BLANK INPUT FILE \n",
+    "#oad.generate_inputs(CONFIGURATION_FILE_NAME, overwrite=True)\n",
+    "INPUT_FILE_NAME = pth.join(DATA_FOLDER, \"problem_inputs.xml\")\n",
+    "oad.variable_viewer(INPUT_FILE_NAME)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4. Run the problem \n",
+    "\n",
+    "Once your configuration is defined and inputs checked, you can run the problem "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "disp_problem = oad.evaluate_problem(CONFIGURATION_FILE_NAME, overwrite=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "oad.variable_viewer(disp_problem.output_file_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 5. Change models \n",
+    "\n",
+    "If you want to solve an other problem, for example the second one, you may have to replace one or several modules. Here again everything happens in the [configuration file](./data/beam_problem_stress.yml).\n",
```

Comment:
Maybe recall the second problem?

Reply:
I think we can be specific: "To solve the second problem, you have to replace the last module..."
(and indeed, maybe put something like: " "To solve the second, stress-based, problem...")

Reply:
This was modified